misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/utils/HttpClientRequest.java

```diff
@@ -0,0 +1,173 @@
+ * This class can be used to send http request.
+ */
+public class HttpClientRequest {
+    private static final Logger LOG = LoggerFactory.getLogger(HttpClientRequest.class);
+    private static final int DEFAULT_READ_TIMEOUT = 30000;
+
+    /**
+     * Send an HTTP POST request to a service.
+     *
+     * @param endpoint - service endpoint
+     * @param postBody - message payload
+     * @param headers  http request headers map
+     * @return - HttpResponse from end point
+     * @throws IOException If an error occurs while sending the GET request
+     */
+    public static HttpResponse doPost(String endpoint, String postBody, Map<String, String> headers)
+            throws IOException {
+        HttpURLConnection urlConnection = null;
+        try {
+            urlConnection = getURLConnection(endpoint);
+            setHeadersAndMethod(urlConnection, headers, "POST");
+            try (OutputStream out = urlConnection.getOutputStream();
+                 Writer writer = new OutputStreamWriter(out, StandardCharsets.UTF_8)) {
+                writer.write(postBody);
+            }
+            return buildResponse(urlConnection);
+        } finally {
+            if (urlConnection != null) {
+                urlConnection.disconnect();
+            }
+        }
+    }
+
+    private static HttpURLConnection getURLConnection(String requestUrl) throws IOException {
+        return getURLConnection(requestUrl, DEFAULT_READ_TIMEOUT);
+    }
+
+    private static HttpURLConnection getURLConnection(String requestUrl, int readTimeout) throws IOException {
+        URL url = new URL(requestUrl);
+        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+        conn.setDoOutput(true);
+        conn.setReadTimeout(readTimeout);
+        conn.setConnectTimeout(15000);
+        conn.setDoInput(true);
+        conn.setUseCaches(true);
+        conn.setAllowUserInteraction(false);
+        return conn;
+    }
+
+    private static Map<String, String> readHeaders(URLConnection urlConnection) {
+        Iterator<String> itr = urlConnection.getHeaderFields().keySet().iterator();
+        Map<String, String> headers = new HashMap();
+        while (itr.hasNext()) {
+            String key = itr.next();
+            if (key != null) {
+                headers.put(key, urlConnection.getHeaderField(key));
+            }
+        }
+        return headers;
+    }
+
+    private static void setHeadersAndMethod(HttpURLConnection conn, Map<String, String> headers, String method)
+            throws ProtocolException {
+        for (Map.Entry<String, String> e : headers.entrySet()) {
+            conn.addRequestProperty(e.getKey(), e.getValue());
+        }
+        conn.setRequestMethod(method);
+    }
+
+    private static HttpResponse buildResponse(HttpURLConnection conn) throws IOException {
+        return buildResponse(conn, defaultResponseBuilder, false);
+    }
+
+    private static HttpResponse buildResponse(HttpURLConnection conn,
```

Comment:
We have only one usage of `buildResponse` function. So why do we need pass `throwError` as a parameter which is always `false`.

Reply:
Removed unwanted parameter